extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosImporter.java

```diff
@@ -202,42 +209,33 @@ long importPhotoBatch(
     final HashMap<String, PhotoModel> uploadTokenToDataId = new HashMap<>();
     final HashMap<String, Long> uploadTokenToLength = new HashMap<>();
 
-    /*
-    TODO: resumable uploads https://developers.google.com/photos/library/guides/resumable-uploads
-    Resumable uploads would allow the upload of larger media that don't fit in memory.  To do this,
-    however, seems to require knowledge of the total file size.
-    */
-    // Upload photos
+    // TODO: resumable uploads https://developers.google.com/photos/library/guides/resumable-uploads
+    //  Resumable uploads would allow the upload of larger media that don't fit in memory.  To do
+    //  this however, seems to require knowledge of the total file size.
     for (PhotoModel photo : photos) {
       try {
-        InputStream inputStream;
-        long bytes;
-        if (photo.isInTempStore()) {
-          final InputStreamWrapper streamWrapper =
-              jobStore.getStream(jobId, photo.getFetchableUrl());
-          bytes = streamWrapper.getBytes();
-          inputStream = streamWrapper.getStream();
-        } else {
-          HttpURLConnection conn = imageStreamProvider.getConnection(photo.getFetchableUrl());
-          final long contentLengthLong = conn.getContentLengthLong();
-          bytes = contentLengthLong != -1 ? contentLengthLong : 0;
-          inputStream = conn.getInputStream();
-        }
+        Pair<InputStream, Long> inputStreamBytesPair =
+            getInputStreamForUrl(jobId, photo.getFetchableUrl(), photo.isInTempStore());
 
         String uploadToken =
-            getOrCreatePhotosInterface(jobId, authData).uploadPhotoContent(inputStream);
-        mediaItems.add(new NewMediaItem(getPhotoDescription(jobId, photo), uploadToken));
+            getOrCreatePhotosInterface(jobId, authData)
+                .uploadPhotoContent(inputStreamBytesPair.getFirst());
```

Comment:
Because the input stream gets exhausted in upload I don't think it should be too much of an issue but it might be nice to make sure we close the input stream afterwards either with a finally statement or with a try-with-resources statement. I think this can help with garbage collection